app/services/credit_notes/refresh_draft_service.rb

```diff
@@ -14,7 +15,16 @@ def call
       return result unless credit_note.draft?
 
       credit_note.applied_taxes.destroy_all
-      credit_note.items.update_all(fee_id: fee.id) # rubocop:disable Rails/SkipsModelValidations
+      credit_note.items.each do |item|
+        item.fee_id = fee.id
+
+        if old_fee_values.any?
+          old_entry = old_fee_values.find { |h| h[:credit_note_item_id] == item.id }
+          item.precise_amount_cents = calculate_item_value(item, old_entry[:fee_amount_cents]) if old_entry
+        end
+
+        item.save(validate: false) # rubocop:disable Rails/SkipsModelValidations
```

Comment:
Any reason to skip the validation since now we are changing the values?

Reply:
We were skipping validations before with `update_all` so I just kept behaviour. Added validation trigger!